apps/admin-dashboard/app/routes/_dashboard.students.$id.gift.tsx

```diff
@@ -0,0 +1,107 @@
+import {
+  type ActionFunctionArgs,
+  json,
+  type LoaderFunctionArgs,
+  redirect,
+} from '@remix-run/node';
+import {
+  Form as RemixForm,
+  useActionData,
+  useNavigate,
+} from '@remix-run/react';
+
+import { db } from '@oyster/db';
+import { Button, Form, Modal, Select } from '@oyster/ui';
+
+import { createGoodyOrder } from '@/modules/goody/goody.core';
+import { Route } from '@/shared/constants';
+import {
+  commitSession,
+  ensureUserAuthenticated,
+  toast,
+} from '@/shared/session.server';
+
+export async function loader({ params, request }: LoaderFunctionArgs) {
+  await ensureUserAuthenticated(request);
+  const student = await getStudent(params.id as string);
+
+  if (!student) {
+    return redirect(Route['/students']);
+  }
+
+  return json({ student });
+}
+
+async function getStudent(id: string) {
+  const row = await db
+    .selectFrom('students')
+    .select(['firstName', 'lastName'])
+    .where('id', '=', id)
+    .execute();
+
+  return row;
+}
+
+export async function action({ params, request }: ActionFunctionArgs) {
+  const session = await ensureUserAuthenticated(request);
+
+  try {
+    await createGoodyOrder(params.id as string);
+
+    toast(session, {
+      message: 'Gift Sent.',
+    });
+
+    return redirect(Route['/students'], {
+      headers: {
+        'Set-Cookie': await commitSession(session),
+      },
+    });
+  } catch (e) {
+    return json({ error: (e as Error).message }, { status: 500 });
+  }
+}
+
+export default function SendGiftPage() {
+  const { error } = useActionData<typeof action>() || {};
+
+  const navigate = useNavigate();
+
+  const giftOptions = [{ id: 0, name: 'DoorDash Gift Card' }];
+
+  function onBack() {
+    navigate(-1);
+  }
+
+  return (
+    <Modal onCloseTo={Route['/students']}>
+      <Modal.Header>
+        <Modal.Title>Send Gift</Modal.Title>
+        <Modal.CloseButton />
+      </Modal.Header>
+
+      <Modal.Description>What gift would you like to send?</Modal.Description>
+
+      <RemixForm className="form" method="post">
+        <Form.ErrorMessage>{error}</Form.ErrorMessage>
+        <Select placeholder="Select a gift...">
+          {giftOptions.map((gift) => {
+            return (
+              <option key={gift.id} value={gift.id}>
+                {gift.name}
+              </option>
+            );
+          })}
+        </Select>
```

Comment:
I'm not sure I fully understand the logic here. It seems that the idea is we can select which gift we want to send (which for now is just a Goody gift), and then once selected we can submit the form. However, once the form is submitted this will trigger the `action` function, which right now will always just try to submit a Goody order.

It might be better to simplify this to only manage the case for a Goody gift right now (so no need for the `RemixForm`). Alternatively, we can handle the logic for the `action` function to only trigger a Goody order if that is the type of gift selected. Right now the main issue I see is just that the form allows us to select a gift, but the `action` function always sends a goody gift. Lmk what you think!

Reply:
Good question @tomas-salgado, here's what @ramiAbdou wrote in the [original](https://github.com/colorstackorg/oyster/issues/187) issue:
> In the Admin Dashboard -> Students page, add an option when clicking on the row's option to "Send Goody Gift". This should open a modal which contains a form asking for the "Gift", which for now should have "DoorDash Gift Card" as the only option.

The [goody api](https://developer.ongoody.com/automation-api/create-an-order) allows you to pass in any product ID from their catalogue to send a gift. ATM we are only sending doordash gift cards, however I've implemented the logic and functionality which will allow us to pass in other product IDs and potentially gift members different gifts from the goody platform. LMK if that makes sense.

Reply:
Ahh I see. I hadn't realized there were different products all from Goody that we might want. In that case this makes a lot more sense and LGTM!